pkg/templateservicebroker/api/types.go

```diff
@@ -0,0 +1,62 @@
+package api
+
+import (
+	templateapi "github.com/openshift/origin/pkg/template/api"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+)
+
+// +genclient=true
+
+type TemplateProvision struct {
+	unversioned.TypeMeta
+	kapi.ObjectMeta
+
+	Spec   TemplateProvisionSpec
+	Status TemplateProvisionStatus
+}
+
+type TemplateProvisionSpec struct {
+	Template  templateapi.Template
```

Comment:
so the intent here is to embed the entire template object as it existed when the provision request was made, with the parameter values (generated or user supplied) in the associated Secret?